src/xtopcom/xblockstore/src/xvblockhub.cpp

```diff
@@ -1629,6 +1653,31 @@ namespace top
                 {
                     m_meta->_highest_genesis_connect_height = this_block_height;
                     m_meta->_highest_genesis_connect_hash   = this_block->get_block_hash();
+                    
```

Comment:
logical structure almost identical to the update of _highest_connect_block_height except set_block_flag.